### PR TITLE
User profile page should show comments and reviews #316

### PR DESCRIPTION
**Pull request by LindanyRetroRabbit**
Though I am not sure about the relationship between Reviewer, ReviewResponse and the UserComment tbl

### CODE REVIEW

File: api/app/users/api.py [hunk 1]
```diff
@@ -545,3 +554,31 @@ def get(self):
...
+
+        current_user = user_repository.get_by_id(g.current_user['id'])
+        if not current_user.is_event_admin(args['event_id']):
+            return FORBIDDEN
+
+        comments = db.session.query(UserComment).filter(
+            UserComment.event_id == args['event_id'],
+            UserComment.user_id == args['user_id']).all()
+        
+        review_by_user_firstname = user_repository.get_by_id(args['user_id']).firstname
```
Comment (jaderabbit) [inline api/app/users/api.py]: So, who is this getting? The current users's firstname? 
Should we not be getting the first name of the reviewers who made the comments?

File: api/app/users/api.py [hunk 2]
```diff
@@ -545,3 +554,31 @@ def get(self):
...
+        
+        reviewer_id = db.session.query(ReviewResponse).filter(
+            ReviewResponse.reviewer_user_id == args['user_id'])
```
Comment (jaderabbit) [inline api/app/users/api.py]: I'm not sure what this code is doing?

Comment (LindanyRetroRabbit) [reply api/app/users/api.py]: image

Comment (LindanyRetroRabbit) [reply api/app/users/api.py]: Creates and  returns an object with 4 attributes.

Comment (jaderabbit) [reply api/app/users/api.py]: But we should not be  filtering by the reviewer_user_id = args['user_id']. 

A specific user creates a specific Response object (application response). (so you'll need to filter on user_id for Response)
ReviewResponse's are assigned to specific Response objects
Those are the reviews you need to pull
Reviewers are assigned to specific ReviewResponse's - reviewer_user_id - get the names of *those* users

And the call to get reviews for a specific user, should perhaps be in the reviews api.py, not under user?

File: api/app/users/api.py [hunk 3]
```diff
@@ -543,5 +548,55 @@ def get(self):
...
+        req_parser.add_argument('user_id', type=int, required=True)
+        args = req_parser.parse_args()
+        user_id = args['user_id']
+        event_id =args['event_id']
+        
+        current_user = user_repository.get_by_id(g.current_user['id'])
+        if not current_user.is_event_admin(args['event_id']):
+            return FORBIDDEN
+
+        comments = db.session.query(UserComment).filter(UserComment.event_id == args['event_id'], UserComment.user_id == args['user_id']).all()
```
Comment (jaderabbit) [inline api/app/users/api.py]: Not every review has a comment, so you can't use it to get the reviewers.
You need to use reviews to get reviewers.

I've advised on this before. Please look at how I explained the logic to work before.

**Comment by LindanyRetroRabbit** [comment]
#316

**Pull request closed without merging**